src/analyses/mHPAnalysis.ml

```diff
@@ -12,7 +12,9 @@ struct
     include MHP
     let name () = "mhp"
     let may_race = MHP.may_happen_in_parallel
-    let should_print _ = true
+    let should_print {tid; created; must_joined} =
+      not (ConcDomain.ThreadSet.is_empty created)
+      || not (ConcDomain.ThreadSet.is_empty must_joined)
```

Comment:
How is this an improvement? All this information is needed to understand why accesses are considered racy.

Reply:
Ok, for must-joined I guess this holds.
An empty created set is very strong information though (no information would be top).

Reply:
If the created set is empty, we're already single-threaded and that's already reflected by the thread flag.